crates/subspace-networking/examples/get-pieces-complex.rs

```diff
@@ -116,27 +116,33 @@ async fn main() {
     };
 
     // create a range from expected peer's public key
-    let from = {
+    let start = {
         let mut buf = peer_id_public_key;
         buf[16] = 0;
         PieceIndexHash::from(buf)
     };
-    let to = {
+    let end = {
         let mut buf = peer_id_public_key;
         buf[16] = 50;
         PieceIndexHash::from(buf)
     };
 
-    let stream_future = node.get_pieces_by_range(from, to);
-    let mut stream = stream_future.await.unwrap();
-    if let Some(value) = stream.next().await {
-        if value != expected_response.pieces {
+    let middle = (U256::from(start) / 2 + U256::from(end) / 2).to_be_bytes();
```

Comment:
This code relates to the farmer now. I would test `get_closest_peers` only here (that it found the correct peer ID) and I would also rename the example.